crates/polars-plan/src/dsl/mod.rs

```diff
@@ -1229,19 +1227,54 @@ impl Expr {
                 move |s| {
                     let mut by = s[1].clone();
                     by = by.rechunk();
-                    let s = &s[0];
+                    let series: Series;
 
                     polars_ensure!(
                         options.weights.is_none(),
                         ComputeError: "`weights` is not supported in 'rolling by' expression"
                     );
-                    let (by, tz) = match by.dtype() {
+                    let (mut by, tz) = match by.dtype() {
                         DataType::Datetime(tu, tz) => {
                             (by.cast(&DataType::Datetime(*tu, None))?, tz)
                         },
                         _ => (by.clone(), &None),
                     };
-                    ensure_sorted_arg(&by, expr_name)?;
+                    let sorting_indices;
+                    let original_indices;
+                    let by_flag = by.is_sorted_flag();
+                    match by_flag {
+                        IsSorted::Ascending => {
+                            series = s[0].clone();
+                            original_indices = None;
+                        },
+                        IsSorted::Descending => {
+                            series = s[0].reverse();
+                            by = by.reverse();
+                            original_indices = None;
+                        },
+                        IsSorted::Not => {
+                            if options.warn_if_unsorted {
+                                eprintln!(
+                                    "PolarsPerformanceWarning: Series is not known to be \
+                                    sorted by `by` column, so Polars is temporarily \
+                                    sorting it for you.\n\
+                                    You can silence this warning by:\n\
+                                    - passing `warn_if_unsorted=False`;\n\
+                                    - sorting your data by your `by` column beforehand;\n\
+                                    - setting `.set_sorted()` if you already know your data is sorted\n\
+                                    before passing it to the rolling aggregation function"
+                                );
+                            }
+                            sorting_indices = by.arg_sort(Default::default());
+                            unsafe { by = by.take_unchecked(&sorting_indices)? };
+                            unsafe { series = s[0].take_unchecked(&sorting_indices)? };
+                            let int_range =
+                                UInt32Chunked::from_iter_values("", 0..s[0].len() as u32)
```

Comment:
This should be `IdxCa` and `IdxSize` instead of `u32`, no? EDIT: see below, you shouldn't save this at all and instead save `sorting_indices`.